### ISSUE TITLE
Bug:`Locale::getDefault()` is different from `$this->request->getLocale()` when $negotiateLocale is true

### ISSUE THREAD

**Issue by kenjis** [codeigniter4/CodeIgniter4#7668]
version: develop (4.3.x)

`Locale::getDefault()` is different from `$this->request->getLocale()` when we use content negotiation.

```diff
--- a/app/Config/App.php
+++ b/app/Config/App.php
@@ -83,7 +83,7 @@ class App extends BaseConfig
      *
      * If false, no automatic detection will be performed.
      */
-    public bool $negotiateLocale = false;
+    public bool $negotiateLocale = true;
 
     /**
      * --------------------------------------------------------------------------
@@ -98,7 +98,7 @@ class App extends BaseConfig
      *
      * @var string[]
      */
-    public array $supportedLocales = ['en'];
+    public array $supportedLocales = ['en', 'ja'];
 
     /**
      * --------------------------------------------------------------------------
```
```php
<?php

namespace App\Controllers;

use Locale;

class Home extends BaseController
{
    public function index()
    {
        return 'Request:' . $this->request->getLocale()
            . ', Locale:' . Locale::getDefault();
    }
}
```
```diff
--- a/app/Config/Routes.php
+++ b/app/Config/Routes.php
@@ -31,6 +31,8 @@ $routes->set404Override();
 // route since we don't have to scan directories.
 $routes->get('/', 'Home::index');
 
+$routes->get('{locale}', 'Home::index');
+
 /*
  * --------------------------------------------------------------------
  * Additional Routing
```
When we use only content negotiation:
```
$ curl -H "Accept-Language: ja,en-US;q=0.9,en;q=0.8" http://localhost:8080/
Request:ja, Locale:en
```
But If `{locale}` is used in a route, both locales are set to `ja`.
```
$ curl -H "Accept-Language: ja,en-US;q=0.9,en;q=0.8" http://localhost:8080/ja
Request:ja, Locale:ja
```

**Comment by michalsn** [comment]
Why is this an issue?

1. The content negotiation example is okay.

2. If `{locale}` is used in the route, then treating it as default is logical. Nobody said that `Locale::getDefault()` has to be equal to `Config\App::defaultLocale`. Current behavior gives a nice option to distinguish lang used in the route and client preference.

**Comment by kenjis** [comment]
@michalsn Do you say the current behavior is expected?

Content negotiation:
Request:ja, Locale:en

`{locale}` in the route:
Request:ja, Locale:ja

**Comment by michalsn** [comment]
@kenjis I think yes.

Is there any other way (method) to find out what `{locale}` is being used? When we specify a language in a URL, it seems reasonable to treat it as a default locale.

**Comment by kenjis** [comment]
@michalsn Okay, I also think the case of `{locale}` in the route is okay.

But why is the `Locale` not changed when using Content Negotiation?

Content negotiation:
Request:ja, Locale:en

**Comment by michalsn** [comment]
@kenjis The framework has no basis to set `jp` as the default `Locale` when we use Content Negotiation.

When the locale is in the URL, then we clearly specify the application to work in that language. So to speak, we overwrite the settings from the config file.

**Comment by kenjis** [comment]
I don't know the difference between the locale in the URL and the locale in Accept-Language header.
When we use Content Negotiation, It seems to me it clearly specify the application to work in that language.

But if the current behavior is correct, we need to fix the behavior in `4.4` branch.

```
(4.4 *$=)$ curl -H "Accept-Language: ja,en-US;q=0.9,en;q=0.8" http://localhost:8080/
Request:ja, Locale:ja
```
```
(4.4 *$=)$ curl -H "Accept-Language: ja,en-US;q=0.9,en;q=0.8" http://localhost:8080/ja
Request:ja, Locale:ja
```

**Comment by michalsn** [comment]
@kenjis My bad. I think you're right. 

`Locale::getDefault()` should always be the same as `$this->request->getLocale()` by default.

**Comment by kenjis** [comment]
@michalsn No problem! If you have any concern about it, feel free to say.

I think if these locale values are different, it is very confusing.

**Comment by kenjis** [comment]
This bug was fixed in 4.4 branch.
